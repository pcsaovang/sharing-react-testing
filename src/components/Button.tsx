import classNames from "classnames";
import Spinner from "./Spinner";

type Props = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  children: React.ReactNode;
  loading?: boolean;
};

function Button({ children, loading, ...props }: Props): React.ReactElement {
  const buttonClass = classNames(
    "w-full",
    "inline-flex",
    "items-center",
    "justify-center",
    "px-4",
    "py-2",
    "bg-red-600",
    "border",
    "border-transparent",
    "rounded-md",
    "font-semibold",
    "capitalize",
    "text-white",
    "hover:bg-red-700",
    "active:bg-red-700",
    "focus:outline-none",
    "focus:border-red-700",
    "focus:ring",
    "focus:ring-red-200",
    "disabled:opacity-25",
    // "transition"
  );
  return (
    <button {...props} className={buttonClass}>
      {loading && <Spinner />}
      {children}
    </button>
  );
}

export default Button;
