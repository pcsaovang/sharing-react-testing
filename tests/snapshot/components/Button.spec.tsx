import { render } from "@testing-library/react";
import Button from "../../../src/components/Button";

describe("Button component", () => {
  it("Render Button with default props", () => {
    const { container } = render(<Button>Click me</Button>);
    expect(container).toMatchSnapshot();
  });

  it("Render Button with loading prop", () => {
    const { container } = render(<Button loading>Click me</Button>);
    expect(container).toMatchSnapshot();
  });
  
  it("Render Button with disabled prop", () => {
    const { container } = render(<Button disabled>Click me</Button>);
    expect(container).toMatchSnapshot();
  });
});

