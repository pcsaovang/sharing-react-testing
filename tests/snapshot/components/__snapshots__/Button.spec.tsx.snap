// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Render Button with default props 1`] = `
<div>
  <button
    class="w-full inline-flex items-center justify-center px-4 py-2 bg-red-600 border border-transparent rounded-md font-semibold capitalize text-white hover:bg-red-700 active:bg-red-700 focus:outline-none focus:border-red-700 focus:ring focus:ring-red-200 disabled:opacity-25"
  >
    Click me
  </button>
</div>
`;

exports[`Button component Render Button with disabled prop 1`] = `
<div>
  <button
    class="w-full inline-flex items-center justify-center px-4 py-2 bg-red-600 border border-transparent rounded-md font-semibold capitalize text-white hover:bg-red-700 active:bg-red-700 focus:outline-none focus:border-red-700 focus:ring focus:ring-red-200 disabled:opacity-25"
    disabled=""
  >
    Click me
  </button>
</div>
`;

exports[`Button component Render Button with loading prop 1`] = `
<div>
  <button
    class="w-full inline-flex items-center justify-center px-4 py-2 bg-red-600 border border-transparent rounded-md font-semibold capitalize text-white hover:bg-red-700 active:bg-red-700 focus:outline-none focus:border-red-700 focus:ring focus:ring-red-200 disabled:opacity-25"
  >
    <svg
      class="animate-spin -ml-1 mr-3 h-5 w-5 text-white"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="opacity-25"
        cx="12"
        cy="12"
        r="10"
        stroke="currentColor"
        stroke-width="4"
      />
      <path
        class="opacity-75"
        d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
        fill="currentColor"
      />
    </svg>
    Click me
  </button>
</div>
`;
